src/main/java/ch/njol/skript/expressions/ExprTool.java

```diff
@@ -102,35 +105,37 @@ public void setItem(final @Nullable ItemStack item) {
 						};
 					}
 				}
-				final EntityEquipment e = p.getEquipment();
-				if (e == null)
+				final EntityEquipment eq = ent.getEquipment();
+				if (eq == null)
 					return null;
-				return new EquipmentSlot(e, EquipmentSlot.EquipSlot.TOOL) {
+				return new EquipmentSlot(eq, offHand ? EquipmentSlot.EquipSlot.OFF_HAND : EquipmentSlot.EquipSlot.TOOL) {
 					@Override
 					public String toString(@Nullable Event event, boolean debug) {
-						return (getTime() == 1 ? "future " : getTime() == -1 ? "former " : "") + Classes.toString(getItem());
+						String time = getTime() == 1 ? "future " : getTime() == -1 ? "former " : "";
```

Comment:
Does time need to be taken into account here too?

Reply:
I thought I replied to this. I think it does because of some events which have a past item.
Im not entirely sure though as I didnt write this originally.